src/TRestRawSignalRecoverChannelsProcess.cxx

```diff
@@ -170,48 +178,150 @@ TRestEvent* TRestRawSignalRecoverChannelsProcess::ProcessEvent(TRestEvent* evInp
+            dataRecovered[n] = 0;
+            dataRight[n] = 0;
+            dataLeft[n] = 0;
+            dataRightClean[n] = 0;
+            dataLeftClean[n] = 0;
+        }
+
+        if (a == 0 && b == 0) {  // if the neigbours are alive
+            if (leftSgnl != nullptr) {
+                for (int n = 0; n < nPoints; n++) {
+                    dataRecovered[n] = leftSgnl->GetData(n);
+                    dataLeft[n] = leftSgnl->GetRawData(n);
+                    dataLeftClean[n] = leftSgnl->GetData(n);
+                }
+            }
 
-        if (leftSgnl != nullptr) {
-            for (int n = 0; n < nPoints; n++) dataRecovered[n] = leftSgnl->GetData(n);
+            if (rightSgnl != nullptr) {
+                for (int n = 0; n < nPoints; n++) {
+                    dataRecovered[n] += rightSgnl->GetData(n);
+                    dataRight[n] = rightSgnl->GetRawData(n);
+                    dataRightClean[n] = rightSgnl->GetData(n);
+                }
+            }
         }
+        /*else{
+            if (leftSgnl2 != nullptr) {
+                for (int n = 0; n < nPoints; n++) dataRecovered[n] = leftSgnl2->GetData(n);
+            }
+            if (rightSgnl2 != nullptr) {
+                for (int n = 0; n < nPoints; n++) dataRecovered[n] += rightSgnl2->GetData(n);
+            }
+        }*/ //This part would be for the case where two dead channels are together, then the recovered channel will be the average between the second-neighbours
 
+        vector<Short_t> offSet(nPoints);
         if (rightSgnl != nullptr) {
-            for (int n = 0; n < nPoints; n++) dataRecovered[n] += rightSgnl->GetData(n);
+            for (int n = 0; n < nPoints; n++)
+                offSet[n] = dataRight[n] - dataRightClean[n];  // In IAXO-D0 both values are the same
+        } else
+            for (int n = 0; n < nPoints; n++)
+                offSet[n] = dataLeft[n] -
+                            dataLeftClean[n];  // Here we set the artifitial baseline of the recovered channel
+
+        if (a == 0 && b == 0) {
+            for (int n = 0; n < nPoints; n++) recoveredSignal->AddPoint(offSet[n] + dataRecovered[n] / 2.);
+        } else {
+            for (int n = 0; n < nPoints; n++) recoveredSignal->AddPoint(offSet[n] + dataRecovered[n] / 2.);
         }
 
-        for (int n = 0; n < nPoints; n++) recoveredSignal->AddPoint(dataRecovered[n] / 2.);
-
         fOutputSignalEvent->AddSignal(*recoveredSignal);
 
-        delete recoveredSignal;
-        /*
-        cout << "Channel recovered!! " << endl;
+        TRestRawSignal* newSignalR = new TRestRawSignal();
+        newSignalR->SetID(idR);
+        TRestRawSignal* newSignalL = new TRestRawSignal();
+        newSignalL->SetID(idL);
+        if (fOutputSignalEvent->GetSignalIndex(idR) > 0) fOutputSignalEvent->RemoveSignalWithId(idR);
+        if (fOutputSignalEvent->GetSignalIndex(idL) > 0) fOutputSignalEvent->RemoveSignalWithId(idL);
+
+        double_t prop = 0.;
```

Comment:
`prop` is only used inside the `for` loop that follows, thus I think it would be more clear to define it inside the `for` loop so that it only has effect inside the loop.

It might solve future problems, and for new people reading the code they will be aware immediately that this variable is only used inside the loop.